dotCMS/src/integration-test/java/com/dotcms/storage/StoragePersistenceAPITest.java

```diff
@@ -0,0 +1,279 @@
+package com.dotcms.storage;
+
+import static com.dotcms.storage.StoragePersistenceProvider.DEFAULT_STORAGE_TYPE;
+import static com.dotcms.unittest.TestUtil.upperCaseRandom;
+import static junit.framework.Assert.assertNotSame;
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertFalse;
+import static junit.framework.TestCase.assertNotNull;
+import static junit.framework.TestCase.assertTrue;
+
+import com.dotcms.datagen.TestDataUtils.TestFile;
+import com.dotcms.storage.StoragePersistenceProvider.INSTANCE;
+import com.dotcms.util.ConfigTestHelper;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.util.Config;
+import com.google.common.collect.ImmutableMap;
+import com.tngtech.java.junit.dataprovider.DataProvider;
+import com.tngtech.java.junit.dataprovider.DataProviderRunner;
+import com.tngtech.java.junit.dataprovider.UseDataProvider;
+import java.io.File;
+import org.apache.commons.lang.RandomStringUtils;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(DataProviderRunner.class)
+public class StoragePersistenceAPITest {
+
+    private static final String textFilePath = "textFilePath";
+    private static final String imageFilePath = "imageFilePath";
+
+    private static final String TEST_IMAGE_JPG = TestFile.JPG.getFilePath();
+    private static final String TEST_IMAGE_PNG = TestFile.PNG.getFilePath();
+    private static final String TEST_TEXT_PATH = TestFile.TEXT.getFilePath();
+
+    private static final StoragePersistenceProvider persistenceProvider = INSTANCE.get();
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```

Comment:
Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)